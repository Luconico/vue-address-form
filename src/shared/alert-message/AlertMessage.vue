<template>
  <div class="alert-container" v-bind:class="alertClass">
    <div class="alert__icon" v-bind:class="inconClass"></div>
    <span class="alert__message">{{ $t(message.value) }}</span>
  </div>
</template>

<script>
import { computed } from "@vue/runtime-core";
export default {
  name: "alert-message",
  props: {
    message: {
      value: {
        type: String,
        required: true,
      },
      msgType: {
        type: String,
        default: "success", // success, error
      },
    },
  },
  setup(props) {
    return {
      alertClass: computed(() =>
        props.message.msgType === "warning" ? "alert--warning" : 
        props.message.msgType === "error" ? "alert--error" : 
        "alert--success"
      ),
      inconClass: computed(() =>
        props.message.msgType === "warning" ? "fa fa-lg fa-exclamation" : 
        props.message.msgType === "error" ? "fa fa-lg fa-ban" : 
        "fa fa-lg fa-check"
      ),
    };
  },
};
</script>

<style>
.alert-container {
  display: flex;
  align-items: center;
  gap: 10px;
  padding: 15px;
  border-radius: 0.4rem;
  margin-bottom: 20px;
}

.alert__icon {
  background-position: center;
  background-repeat: no-repeat;
  background-size: contain;
  height: 20px;
  width: 20px;
}

.alert--success {
  background-color: #c8f8c9;
  color: #008a00;
}
.alert--error {
  background-color: #f8d0c8;
  color: #900;
}
</style>
