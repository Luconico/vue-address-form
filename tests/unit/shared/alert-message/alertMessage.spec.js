import AlertMessage from '@/shared/alert-message/AlertMessage.vue';
import { shallowMount } from '@vue/test-utils';
import { config } from "@vue/test-utils"
config.global.mocks = {
  $t: (msg) => msg
}
describe('AlertMessage.vue', () => {
  
  const propsData = {
    message: {
      value: 'test',
      msgType: 'success'
    }
  };
  const wrapper = shallowMount(AlertMessage, { propsData } );

  it('Should match whit snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Should props message', () => {
    expect(wrapper.props()).toEqual(propsData);
  });

  
});

