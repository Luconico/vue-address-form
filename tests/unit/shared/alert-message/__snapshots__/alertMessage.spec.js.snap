// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertMessage.vue Should match whit snapshot 1`] = `
<div class="alert-container alert--success">
  <div class="alert__icon fa fa-lg fa-check"></div><span class="alert__message">test</span>
</div>
`;
